### PR TITLE
[SDK-3135] New Express API Interactive QuickStart

### PR DESCRIPTION
**Pull request by adamjmcgrath**
image

### CODE REVIEW

**Review by frederikprijck — DISMISSED** [review]

**Review by frederikprijck — COMMENTED** [review]
Should this need a checkpoint section as mentioned in https://oktawiki.atlassian.net/wiki/spaces/DXSDK/pages/2540963444/How+to+Migrate+a+Quickstart+to+the+New+Design#Checkpoints ?

Using a checkpoint is the new way to gather feedback from the user regarding the quickstart.

File: articles/quickstart/backend/nodejs/interactive.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+```bash
+npm install --save express-oauth2-jwt-bearer
+```
+
+## Configure the middleware {{{ data-action=code data-code="server.js#1:10" }}}
+
+Configure `express-oauth2-jwt-bearer` with your Domain and API Identifier.
+
+The `checkJwt` middleware shown to the right checks if the user's Access Token included in the request is valid. If the token is not valid, the user gets a 401 Authorization error when they try to access the endpoints.
```
Comment (stevehobbsdev) [inline articles/quickstart/backend/nodejs/interactive.md]: Style guide (word list): access token should be lowercase

```suggestion
The `checkJwt` middleware shown to the right checks if the user's access token included in the request is valid. If the token is not valid, the user gets a 401 Authorization error when they try to access the endpoints.
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+## Protect API endpoints {{{ data-action=code data-code="server.js#12:32" }}}
+
+To protect an individual route by requiring a valid JWT, configure the route with the `checkJwt` middleware constructed from `express-oauth2-jwt-bearer`.
+
+You can configure individual routes to look for a particular scope. To achieve that, set up another middleware with the `requiresScope` method. Provide the required scopes and apply the middleware to any routes you want to add authorization to.
+
+Pass the `checkJwt` and `requiredScopes` middlewares to the route you want to protect.
+
+In this configuration, only Access Tokens with the `read:messages` scope can access the endpoint.
```
Comment (stevehobbsdev) [inline articles/quickstart/backend/nodejs/interactive.md]: Style guide (word list): access token should be lowercase

```suggestion
In this configuration, only access tokens with the `read:messages` scope can access the endpoint.
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+<%= include('../_includes/_call_api') %>
+
+::::checkpoint
+:::checkpoint-default
+Now that you have configured your application, run your application to verify that:
+* `GET /api/public` is available for non-authenticated requests.
+* `GET /api/private` is available for authenticated requests.
+* `GET /api/private-scoped` is available for authenticated requests containing an Access Token with the `read:messages` scope.
```
Comment (stevehobbsdev) [inline articles/quickstart/backend/nodejs/interactive.md]: Style guide (word list): access token should be lowercase

```suggestion
* `GET /api/private-scoped` is available for authenticated requests containing an access token with the `read:messages` scope.
```

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]
LGTM, just left some style guide-related thing

**Review by stevehobbsdev — DISMISSED** [review]

**Review by frederikprijck — DISMISSED** [review]

File: articles/quickstart/backend/nodejs/interactive.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+files:
+- files/server
+---
+
+<!-- markdownlint-disable MD041 MD025 -->
+
+# Add authorization to an Express.js API application
+This guide demonstrates how to integrate Auth0 with any new or existing Express.js API application using the `express-oauth2-jwt-bearer` package.
+
+If you haven't created an API in your Auth0 dashboard yet, you can use the interactive selector to create a new Auth0 API or select an existing API that represents the project you want to integrate with.
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
If you have not created an API in your Auth0 dashboard yet, use the interactive selector to create a new Auth0 API or select an existing project API.
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+<!-- markdownlint-disable MD041 MD002 -->
+
+<%= include('../../../_includes/_api_auth_intro') %>
+
+## Define permissions
+<%= include('../_includes/_api_scopes_access_resources') %>
+
+## Install dependencies
+
+First install the SDK using npm.
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
First, install the SDK with `npm`.
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+npm install --save express-oauth2-jwt-bearer
+```
+
+## Configure the middleware {{{ data-action=code data-code="server.js#1:10" }}}
+
+Configure `express-oauth2-jwt-bearer` with your Domain and API Identifier.
+
+The `checkJwt` middleware shown to the right checks if the user's access token included in the request is valid. If the token is not valid, the user gets a 401 Authorization error when they try to access the endpoints.
+
+The middleware doesn't check if the token has the sufficient scope to access the requested resources.
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
The middleware does not check if the token has sufficient scope to access the requested resources.
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+:::checkpoint-default
+Now that you have configured your application, run your application to verify that:
+* `GET /api/public` is available for non-authenticated requests.
+* `GET /api/private` is available for authenticated requests.
+* `GET /api/private-scoped` is available for authenticated requests containing an access token with the `read:messages` scope.
+:::
+
+:::checkpoint-failure
+Sorry about that. Here's a couple things to double check:
+* make sure the token is added as the `Authorization` header
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
If your application did not start successfully:
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+* does the token have the correct scopes? You can use [jwt.io](https://jwt.io/) to verify.
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
* Verify you added the token as the `Authorization` header
* Ensure the token has the correct scopes. Verify with [jwt.io](https://jwt.io/).
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+<!-- markdownlint-disable MD041 MD025 -->
+
+# Add authorization to an Express.js API application
+This guide demonstrates how to integrate Auth0 with any new or existing Express.js API application using the `express-oauth2-jwt-bearer` package.
+
+If you haven't created an API in your Auth0 dashboard yet, you can use the interactive selector to create a new Auth0 API or select an existing API that represents the project you want to integrate with.
+
+Alternatively, you can read [our getting started guide](get-started/auth0-overview/set-up-apis) that helps you set up your first API through the Auth0 dashboard.
+
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
To set up your first API through the Auth0 dashboard, review [our getting started guide](get-started/auth0-overview/set-up-apis).
```

File: articles/quickstart/backend/nodejs/interactive.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+Every API in Auth0 is configured using an API Identifier that your application code will use as the Audience to validate the access token.
```
Comment (tr0njavolta) [inline articles/quickstart/backend/nodejs/interactive.md]:
```suggestion
Each Auth0 API uses the API Identifier, which your application needs to validate the access token.
```

**Review by tr0njavolta — CHANGES_REQUESTED** [review]
Some suggestions and style updates.

**Review by tr0njavolta — APPROVED** [review]

**Pull request merged**